lib/console.js

```diff
@@ -35,20 +37,48 @@ function Console(stdout, stderr) {
   }
 }
 
+function write(ignoreErrors, stream, string) {
+  if (!ignoreErrors) return stream.write(string);
+
+  // There may be an error occurring synchronously (e.g. for files or TTYs)
+  // or asynchronously (e.g. pipes), so handle both situations.
+  try {
+    // Add and later remove a noop error handler to catch synchronous errors.
+    stream.once('error', noop);
+
+    stream.write(string, (err) => {
```

Comment:
It can’t be top-level because it still needs access to the stream but I’ve attached these to the console object itself, does that work for you? It definitely avoids the closure problem.